__tests__/util.test.ts

```
@@ -162,3 +164,33 @@ test("getClubhouseURLFromPullRequest comment", async () => {
 
   scope.done();
 });
+
+test("shouldProcessPullRequestForUser user lists not defined", async () => {
+  expect(util.shouldProcessPullRequestForUser("github-author-1")).toBeTruthy;
+});
+
+test("shouldProcessPullRequestForUser both user lists defined", async () => {
+  process.env["INPUT_ONLY-USERS"] = "fake-user-1";
+  process.env["INPUT_IGNORED-USERS"] = "fake-user-1";
+  expect(util.shouldProcessPullRequestForUser("fake-user-1")).toBeFalsy;
```

This should throw an exception.

---

If I simply remove the `return false`, but keep the `core.setFailed` that should meet the criteria correct? Referencing [GH Action documentation](https://docs.github.com/en/actions/creating-actions/setting-exit-codes-for-actions#setting-a-failure-exit-code-in-a-javascript-action) as I wanted to see how to correctly throw an error.

---

`core.setFailed` will set metadata that GitHub Actions uses to determine if the run was successful or not. However, it does not cause the run to halt. If you don't throw an error, it's possible that something in this code will continue to process this pull request in a way we don't want.

Also, conceptually it doesn't make sense to return true _or_ false if we have conflicting information, so we should throw an error instead.

---

Fixed.